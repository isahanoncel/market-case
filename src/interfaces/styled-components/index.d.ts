// #region Global Imports
import 'styled-components';
// #endregion Global Imports

export type Icons = 'Bag' | 'ArrowLeft' | 'Check';

export type Colors =
  | 'primary'
  | 'gray'
  | 'secondary'
  | 'white'
  | 'background'
  | 'transparent'
  | 'innerBorder'
  | 'black100'
  | 'black600';

type Fonts = 'OpenSans';

declare module 'styled-components' {
  export interface DefaultTheme {
    colors: Record<Colors, string>;
    fonts: Record<Fonts, string>;
  }
}
