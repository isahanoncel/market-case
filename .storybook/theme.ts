import { create } from '@storybook/theming';

export default create({
  base: 'light',
  brandTitle: 'Getir',
  brandUrl: 'https://getir.com',
  brandTarget: '_self',
});
