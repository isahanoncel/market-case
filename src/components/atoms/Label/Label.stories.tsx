import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import Label from '.';

export default {
  title: 'Atoms/Label',
  component: Label,
} as ComponentMeta<typeof Label>;

export const Default: ComponentStory<typeof Label> = (args) => (
  <Label {...args}>Getir</Label>
);
