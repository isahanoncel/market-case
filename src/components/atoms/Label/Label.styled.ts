import styled from 'styled-components';
import { Colors } from '../../../interfaces/styled-components';

export interface ITitle {
  weight: number;
  size: number;
  lineHeight: number;
  color: Colors;
  margin: string;
}

export const StyledLabel = styled.span<ITitle>`
  font-size: ${({ size }) => size}px;
  line-height: ${({ lineHeight }) => lineHeight}px;
  color: ${({ theme, color }) => theme.colors[color]};
  font-weight: ${({ weight }) => weight};
  margin: ${({ margin }) => margin};
`;
