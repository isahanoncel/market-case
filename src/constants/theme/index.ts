import { DefaultTheme } from 'styled-components';

export const COLORS = {
  white: '#fff',
  primary: '#1EA4CE',
  whiteBlue: '#F2F0FD',
  secondary: '#147594',
  black100: '#E0E0E0',
  black500: '#6F6F6F',
  black600: '#525252',
  gray: '#697488',
  background: '#E5E5E5',
  transparent: 'transparent',
  innerBorder: '#DFDEE2',
  thinGray: '#A8A8A8',
};

export const FONTS = {
  OpenSans: "'Open Sans', sans-serif",
};

const heroTheme: DefaultTheme = {
  colors: COLORS,
  fonts: FONTS,
};

export default heroTheme;
