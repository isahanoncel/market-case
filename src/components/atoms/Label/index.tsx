import React, { FC } from 'react';
import * as S from './Label.styled';
import { Colors } from '../../../interfaces/styled-components';

interface ITitle {
  weight?: number;
  size?: number;
  lineHeight?: number;
  color?: Colors;
  children?: React.ReactNode | string;
  margin?: string;
}

const Label: FC<ITitle> = ({
  weight = 400,
  size = 14,
  lineHeight = 14,
  color = 'black600',
  margin = '0',
  children,
}) => (
  <S.StyledLabel
    color={color}
    weight={weight}
    size={size}
    margin={margin}
    lineHeight={lineHeight}
  >
    {children}
  </S.StyledLabel>
);

export default Label;
