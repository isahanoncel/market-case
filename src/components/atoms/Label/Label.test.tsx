import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { render, screen } from '@testing-library/react';
import Label from '.';
import theme, { COLORS } from '../../../constants/theme';

test('Label snapshot testing', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <Label color="black600" size={22} weight={600}>
          Getir Label
        </Label>
      </ThemeProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Label styles testing', () => {
  render(
    <ThemeProvider theme={theme}>
      <Label color="black600" size={29} weight={600}>
        Getir
      </Label>
    </ThemeProvider>,
  );
  expect(screen.getByText(/Getir/i)).toHaveStyle(
    `color: ${COLORS.black600};font-weight:600;font-size:29px`,
  );
});
