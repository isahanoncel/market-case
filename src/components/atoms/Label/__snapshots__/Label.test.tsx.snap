// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label snapshot testing 1`] = `
.c0 {
  font-size: 22px;
  line-height: 14px;
  color: #525252;
  font-weight: 600;
  margin: 0;
}

<span
  className="c0"
  color="black600"
  size={22}
>
  Getir Label
</span>
`;
